File: app/src/main/java/com/cyl/musiclake/ui/music/mv/MvDetailContract.java
package com.cyl.musiclake.ui.music.mv;

import com.cyl.musicapi.netease.CommentsItemInfo;
import com.cyl.musicapi.netease.MvInfoDetail;
import com.cyl.musicapi.netease.MvInfoDetailInfo;
import com.cyl.musiclake.bean.MvInfoBean;
import com.cyl.musiclake.ui.base.BaseContract;

import java.util.List;


public interface MvDetailContract {

    interface View extends BaseContract.BaseView {
        void showMvList(List<MvInfoDetail> mvList);

        void showBaiduMvDetailInfo(MvInfoBean mvInfoBean);

        void showMvDetailInfo(MvInfoDetailInfo mvInfoDetailInfo);

        void showMvHotComment(List<CommentsItemInfo> mvHotCommentInfo);

        void showMvComment(List<CommentsItemInfo> mvCommentInfo);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {
        void loadMv(int offset);

        void loadMvDetail(String mvid);

        void loadBaiduMvInfo(String songId);

        void loadSimilarMv(String mvid);

        void loadMvComment(String mvid);
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/mv/MvDetailContract.kt
package com.cyl.musiclake.ui.music.mv

import com.cyl.musicapi.netease.CommentsItemInfo
import com.cyl.musicapi.netease.MvInfoDetail
import com.cyl.musicapi.netease.MvInfoDetailInfo
import com.cyl.musiclake.bean.MvInfoBean
import com.cyl.musiclake.ui.base.BaseContract
import com.cyl.musiclake.ui.base.BaseContract.BaseView

interface MvDetailContract {
    interface View : BaseView {
        fun showMvList(mvList: List<MvInfoDetail>)
        fun showBaiduMvDetailInfo(mvInfoBean: MvInfoBean?)
        fun showMvDetailInfo(mvInfoDetailInfo: MvInfoDetailInfo?)
        fun showMvUrlInfo(mvUrl: String?)
        fun showMvHotComment(mvHotCommentInfo: List<CommentsItemInfo>)
        fun showMvComment(mvCommentInfo: List<CommentsItemInfo>)
    }

    interface Presenter : BaseContract.BasePresenter<View?> {
        fun loadMv(offset: Int)
        fun loadMvDetail(mvid: String?)
        fun loadBaiduMvInfo(songId: String?)
        fun loadSimilarMv(mvid: String?)
        fun loadMvComment(mvid: String?, offset: Int)
    }
}